modules/ppcp-api-client/src/Repository/class-applicationcontextrepository.php

```diff
@@ -59,4 +59,20 @@ public function current_context(
 		);
 		return $context;
 	}
+
+	/**
+	 * Returns a PayPal-supported BCP-47 code, for example de-DE-formal becomes de-DE.
+	 *
+	 * @return string
+	 */
+	protected function valid_bcp47_code() {
+		$locale = str_replace( '_', '-', get_user_locale() );
+		$parts  = explode( '-', $locale );
+
+		if ( count( $parts ) < 3 ) {
+			return $locale;
+		}
+
+		return substr( $locale, 0, strrpos( $locale, '-' ) );
+	}
```

Comment:
I have some doubts about this. I believe this will work in most cases but I expect it fail when `get_user_locale()` returns locale containing 3 letters with no delimiter (like `bel` when Belarusian site language is chosen,  also `ceb` for Cebuano, `ary`, `dsb`,  etc.).

PayPal documentation [states](https://developer.paypal.com/docs/api/orders/v2/#definition-order_application_context) they use the `^[a-z]{2}(?:-[A-Z][a-z]{3})?(?:-(?:[A-Z]{2}))?$` regex to validate the language. I tested a few of possible WP locales and it worked for the `xx-XX` format (like `de-DE`), also for `xx` format (like `ar`), but not for `xxx` format (like the `bel` and others I provided above). This function will return three-letters code as is (which is valid BCP 47, if I understand it correctly). But it doesn't meet the regex from PayPal docs provided above. 

I have no good suggestion about this issue. Probably, it would make sense to use that regex to validate the result and fallback to default if validation wasn't passed.